src/test/java/redis/clients/jedis/JedisTest.java

```diff
@@ -288,4 +290,41 @@ public void checkDisconnectOnQuit() {
     assertFalse(jedis.isConnected());
   }
 
+  @Test
+  public void clientSetInfoDefault() {
+    try (Jedis jedis = new Jedis(hnp, DefaultJedisClientConfig.builder().password("foobared")
+        .build())) {
+      assertEquals("PONG", jedis.ping());
+      String info = jedis.clientInfo();
+      assertTrue(info.contains("lib-name=" + JedisMetaInfo.getArtifactId()));
+      assertTrue(info.contains("lib-ver=" + JedisMetaInfo.getVersion()));
+    }
+  }
+
+  @Test
+  public void clientSetInfoDisable() {
+    try (Jedis jedis = new Jedis(hnp, DefaultJedisClientConfig.builder().password("foobared")
+        .clientSetInfoConfig(new ClientSetInfoConfig() {
+          @Override public boolean disable() { return true; }
+        }).build())) {
+      assertEquals("PONG", jedis.ping());
+      String info = jedis.clientInfo();
+      assertFalse(info.contains("lib-name=" + JedisMetaInfo.getArtifactId()));
+      assertFalse(info.contains("lib-ver=" + JedisMetaInfo.getVersion()));
+    }
+  }
+
+  @Test
+  public void clientSetInfoCustom() {
+    final String libNameSuffix = "-for-redis";
+    ClientSetInfoConfig setInfoConfig = DefaultJedisClientConfig.setInfoBuilder()
+        .libNameSuffix(libNameSuffix).build();
+    try (Jedis jedis = new Jedis(hnp, DefaultJedisClientConfig.builder().password("foobared")
+        .clientSetInfoConfig(setInfoConfig).build())) {
+      assertEquals("PONG", jedis.ping());
+      String info = jedis.clientInfo();
+      assertTrue(info.contains("lib-name=" + JedisMetaInfo.getArtifactId() + libNameSuffix));
```

Comment:
When a developer sets `libNameSuffix`, `lib-name` sent to the Redis should look like `jedis(Spring_Data_Redis_v3.1.1)`